pkg/upgrade/initialize.go

```diff
@@ -6,25 +6,15 @@ import (
 	"encoding/json"
 
 	"github.com/openshift/openshift-azure/pkg/api"
-	"github.com/openshift/openshift-azure/pkg/util/azureclient"
 	"github.com/openshift/openshift-azure/pkg/util/azureclient/storage"
 )
 
 func (u *simpleUpgrader) InitializeCluster(ctx context.Context, cs *api.OpenShiftManagedCluster) error {
-	if u.accountsClient == nil {
-		authorizer, err := azureclient.NewAuthorizerFromContext(ctx)
-		if err != nil {
-			return err
-		}
-
-		u.accountsClient = azureclient.NewAccountsClient(cs.Properties.AzProfile.SubscriptionID, authorizer, u.pluginConfig.AcceptLanguages)
-	}
 	if u.storageClient == nil {
 		keys, err := u.accountsClient.ListKeys(ctx, cs.Properties.AzProfile.ResourceGroup, cs.Config.ConfigStorageAccount)
 		if err != nil {
 			return err
 		}
-
 		u.storageClient, err = storage.NewClient(cs.Config.ConfigStorageAccount, *(*keys.Keys)[0].Value, storage.DefaultBaseURL, storage.DefaultAPIVersion, true)
```

Comment:
Shouldn't you be moving the storage client setup out of InitializeCluster too?

Reply:
the sa is created in the deploy, and this is the only place it is used so it kinda makes sense to me.

Reply:
Makes sense. Maybe it would be clearer if we could separate between create and update in createClients.